conans/test/integration/package_id/test_validate.py

```diff
@@ -64,17 +64,43 @@ def package_id(self):
         self.assertIn("pkg/0.1: Package 'cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31' created",
                       client.out)
 
-        client.run("create . pkg/0.1@ -s os=Windows")
-        self.assertIn("pkg/0.1: Main binary package 'INVALID' missing. "
-                      "Using compatible package 'cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31'",
-                      client.out)
-        self.assertIn("pkg/0.1:cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31 - Cache", client.out)
+        client.run("create . pkg/0.1@ -s os=Windows", assert_error=True)
```

Comment:
So now `ConanInvalidConfiguration` doesn't fallback to compatible packages?

Reply:
I need some clarification about this one as well

Reply:
Yes, it is not possible to build a compatible configuration. Compatible configurations, both with ``compatible_packages`` and with deletion of ``self.info.settings`` in ``package_id()`` should exist in order to use them, but they are not built as a result of a fallback.

Note that the previous behavior was really confusing, we did a ``conan create`` for one configuration and the package was not built at all. Note that if you did this command as a result of changing some source code, it could be ignored. A ``conan create`` without changing the build policy should always do a package build.